cpp/arcticdb/storage/s3/s3_storage-inl.hpp

```diff
@@ -124,18 +144,8 @@ void do_update_impl(
         const std::string& bucket_name,
         S3ClientType& s3_client,
         KeyBucketizer&& bucketizer) {
-    // s3 updates the key if it already exists (our buckets don't have versioning)
-    do_write_impl(std::move(kvs), root_folder, bucket_name, s3_client, std::move(bucketizer));
-}
-
-struct UnexpectedS3ErrorException : public std::exception {};
-
-inline bool is_expected_error_type(Aws::S3::S3Errors err) {
-    return err == Aws::S3::S3Errors::NO_SUCH_KEY
-        || err == Aws::S3::S3Errors::NO_SUCH_BUCKET
```

Comment:
We've lost the special handling for `NO_SUCH_BUCKET` in this PR, I assume because you check it's accessible when the Arctic instance is created?

What if the bucket is deleted _after_ the Arctic instance is created?